shell/src/main/java/org/crsh/lang/impl/java/JavaCompiler.java

```diff
@@ -53,7 +54,7 @@
   }
 
   public CommandResolution compileCommand(String name, byte[] source) throws CommandException, NullPointerException {
-    String script = new String(source);
+    String script = new String(source, Charset.forName("UTF-8"));
```

Comment:
It would be better if this was settable, rather than forcing an encoding, especially if the default charset is not desired.